### PR TITLE
Upgrade FastAPI and dependencies

### PR DESCRIPTION
**Pull request by ammar92**
## Changes
Updates FastAPI due to security concerns in its dependencies. Also fixes https://github.com/minvws/nl-kat-coordination/issues/964.

## Checklist for author(s):
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [x] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [x] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [ ] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [x] I have performed a self-review of my own code;
- [ ] I have commented my code, particularly in hard-to-understand areas;
- [ ] I have made corresponding changes to the documentation, if necessary;
- [x] I have written unit, integration, and end-to-end tests for the change that I made;

If a non-trivial PR:
- [x] This PR is part of a milestone and has appropriate labels;
- [ ] This PR is properly linked to the project board (either directly or via an issue);
- [ ] I have added screenshots or some other proof that my code does what it is supposed to do;

### CODE REVIEW

File: boefjes/Dockerfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ ENV PATH=/home/nonroot/.local/bin:${PATH}
 ARG ENVIRONMENT
 
 COPY --from=boefjes-requirements /tmp/boefjes-requirements.txt /tmp/boefjes-requirements.txt
-COPY boefjes/requirements-dev.txt .
+COPY boefjes/requirements-dev.txt boefjes/requirements.txt .
```
Comment (dekkers) [inline boefjes/Dockerfile:25]: boefjes/requirements.txt is already part of /tmp/boefjes-requirements.txt. Also the COPY here isn't very useful if we don't do something with this later?

**Review by dekkers — COMMENTED** [review]
Looks good to me, moving to QA

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* No apparent regressions

**Review by Darwinkel — APPROVED** [review]

File: boefjes/Dockerfile (hunk 1 above)
Comment (Donnype) [reply boefjes/Dockerfile:25]: I added a `-r requirements.txt` in the `requirements-dev.txt` to be more consistent with the other services, hence the extra COPY

File: boefjes/requirements.txt [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 jsonschema==4.17.0
 pydantic==1.10.7
 docker==6.1.2
 requests==2.30.0
-fastapi==0.86.0
-uvicorn==0.19.0
-psycopg2==2.9.6
+fastapi==0.93.0
+uvicorn==0.20.0
+psycopg2-binary==2.9.6
```
[the remark this answers is not in the document]
Comment (dekkers) [reply boefjes/requirements.txt:11]: The problem we ran into has been fixed, so not sure whether should stay with `psycopg2` or switch to `psycopg2-binary`?

[the remark this answers is not in the document]
Comment (ammar92) [reply boefjes/requirements.txt:11]: If the earlier problem has been solved and it's already implemented like that in the other modules then I'm fine with reverting this specific change

**Review by underdarknl — APPROVED** [review]

**Pull request merged**